src/main/java/core/mvc/tobe/ControllerScanner.java

```diff
@@ -0,0 +1,63 @@
+package core.mvc.tobe;
+
+import core.annotation.web.Controller;
+import org.reflections.Reflections;
+import org.reflections.scanners.SubTypesScanner;
+import org.reflections.scanners.TypeAnnotationsScanner;
+import org.reflections.util.ConfigurationBuilder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.InvocationTargetException;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * @author KingCjy
+ */
+public class ControllerScanner {
+
+    private static final Logger logger = LoggerFactory.getLogger(ControllerScanner.class);
+
+    private Reflections reflections;
+
+    public ControllerScanner(Object... basePackage) {
+        this.reflections = new Reflections(ConfigurationBuilder.build(basePackage).setScanners(
+                new SubTypesScanner(false),
+                new TypeAnnotationsScanner()
+        ));
+    }
+
+    public Map<Class<?>, Object> getInstantiateControllers() {
+        Map<Class<?>, Object> result = new LinkedHashMap<>();
+        Set<Class<?>> controllers = findControllers();
+
+        for (Class<?> controller : controllers) {
+            result.put(controller, createInstance(controller));
+        }
+
+        return result;
+    }
+
+    private Set<Class<?>> findControllers() {
+        return reflections.getTypesAnnotatedWith(Controller.class);
+    }
+
+    private Object createInstance(Class<?> targetClass) {
+        try {
+            Constructor constructor = targetClass.getConstructor();
+            return constructor.newInstance();
+        } catch (NoSuchMethodException e) {
+            logger.error(e.getMessage(), e);
+        } catch (IllegalAccessException e) {
+            logger.error(e.getMessage(), e);
```

Comment:
예외 종류에 상관없이 같은 처리를 하고 있네요. 동일한 처리를 한다면 예외를 같이 묶어주는 방법도 있습니다.
예외 처리 방법에 대해서도 아래 링크를 참조해서 한번 고민해보는 것도 좋을 것 같아요 :)
http://www.nextree.co.kr/p3239/

```suggestion
        } catch (NoSuchMethodException | IllegalAccessException e) {
            logger.error(e.getMessage(), e);
        }
```